api/src/main/java/ai/djl/util/ZipUtils.java

```diff
@@ -52,12 +52,10 @@ public static void unzip(InputStream is, Path dest) throws IOException {
         ZipEntry entry;
         Set<String> set = new HashSet<>();
         while ((entry = zis.getNextEntry()) != null) {
-            String name = removeLeadingFileSeparator(entry.getName());
-            if (name.contains("..")) {
-                throw new IOException("Malicious zip entry: " + name);
-            }
-            set.add(name);
-            Path file = dest.resolve(name).toAbsolutePath();
+            String entryName = entry.getName();
+            validateArchiveEntry(entry.getName(), dest);
+            set.add(entryName);
+            Path file = dest.resolve(entryName).toAbsolutePath();
```

Comment:
A bit subjective, feel free to reject:

I know I was the one who said you can make this void, but what if we return the absolute path instead?

```
Path file = getAbsolutePathInDestination(entry.getName(), dest);
set.add(file.toString());
```

```
static Path getAbsolutePathInDestination(String name, Path destination) throws IOException {
    Path outputPath = destination.resolve(name).normalize();
    if (!outputPath.startsWith(destination.normalize())) {
        throw new IOException(
                "Bad archive entry "
                        + name
                        + ". Attempted write outside destination "
                        + destination);
    }
    return outputPath;
}
```